lightly/utils/optim.py

```diff
@@ -0,0 +1,90 @@
+from typing import Any, Dict, Iterable, Optional
+
+from torch.optim import Optimizer
+
+
+def update_param_groups(
+    optimizer: Optimizer,
+    default_update: Optional[Dict[str, Any]] = None,
+    updates: Optional[Iterable[Dict[str, Any]]] = None,
+) -> None:
+    """Update the values in the param groups of the optimizer.
+
+    This is useful to update values following a schedule during training.
+
+    Args:
+        optimizer:
+            A PyTorch optimizer.
+        default_update:
+            Key value pairs that will be updated for all param groups. The value is only
+            updated if the respective key already exists in the param group.
+        updates:
+            A list of dicts with key value pairs. Each dict must contain a key "name"
+            that specifies the name of the param group(s) that should be updated. All
+            values of the dict will be updated in the param group(s) with the same name.
+
+    Examples:
+        >>> optimizer = torch.optim.SGD(
+        >>>     [
+        >>>         {
+        >>>             "name": "model",
+        >>>             "params": params,
+        >>>         },
+        >>>         {
+        >>>             "name": "model_no_weight_decay",
+        >>>             "params": params_no_weight_decay,
+        >>>             "weight_decay": 0.0,
+        >>>         },
+        >>>         {
+        >>>             "name": "head",
+        >>>             "params": head_params,
+        >>>             "weight_decay": 0.5,
+        >>>         },
+        >>>     ],
+        >>>     lr=0.1,
+        >>>     weight_decay=0.2,
+        >>> )
+        >>>
+        >>> update_param_groups(
+        >>>     optimizer=optimizer,
+        >>>     default_update={"weight_decay": 0.3},
+        >>>     updates=[
+        >>>         {"name": "model_no_weight_decay", "weight_decay": 0.0},
+        >>>         {"name": "head", "weight_decay": 0.6},
+        >>>     ]
+        >>> )
+        >>>
+        >>> # Param group "model" has now weight decay 0.3
```

Comment:
According to the docstring this is incorrect because `model` doesn't have a `weight_decay` in it's param group before the update.

Edit: My comment is wrong because we set a `weight_decay` default value.